src/NuGet.Core/NuGet.Configuration/Proxy/ProxyCache.cs

```diff
@@ -99,6 +99,14 @@ public WebProxy GetUserConfiguredProxy()
                 {
                     webProxy.Credentials = new NetworkCredential(userName, password);
                 }
+                
+                var noProxy = _settings.GetValue(SettingsUtility.ConfigSection, ConfigurationContants.NoProxy);
```

Comment:
About the split, it makes sense. I was thinking about one method that takes a no_proxy string and applies it to the webproxy, but I agree now that it is superfluous.

LGTM otherwise.

How did you test this change?